File: app/src/main/java/com/example/cinemates/ui/fragment/MediaImagesFragment.java
package com.example.cinemates.ui.fragment;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProvider;

import com.example.cinemates.adapter.ImageRecyclerViewAdapter;
import com.example.cinemates.databinding.FragmentMediaImagesBinding;
import com.example.cinemates.model.Images;
import com.example.cinemates.model.Movie;
import com.example.cinemates.viewmodel.MovieViewModel;

/**
 * @author Antonio Di Nuzzo
 * Created 19/06/2022 at 09:54
 */
public class MediaImagesFragment extends Fragment {
    private FragmentMediaImagesBinding mBinding;
    private MovieViewModel mViewModel;
    private ImageRecyclerViewAdapter mPosterAdapter, mBackdropAdapter;


    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mViewModel = new ViewModelProvider(getActivity()).get(MovieViewModel.class);
        mPosterAdapter = new ImageRecyclerViewAdapter(getContext());
        mBackdropAdapter = new ImageRecyclerViewAdapter(getContext());
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        mBinding = FragmentMediaImagesBinding.inflate(inflater, container, false);
        Movie movie = (Movie) getArguments().getSerializable("movie");
        mBinding.postersRv.setAdapter(mPosterAdapter);
        mBinding.postersRv.setEmptyView(mBinding.emptyPosterView.getRoot());
        mBinding.backdropRv.setEmptyView(mBinding.emptyBackdropView.getRoot());
        mBinding.backdropRv.setAdapter(mBackdropAdapter);

        mViewModel.getImages().observe(getViewLifecycleOwner(), new Observer<Images>() {
            @Override
            public void onChanged(Images images) {
                mPosterAdapter.addItems(images.getPosters());
                mBackdropAdapter.addItems(images.getBackdrops());
            }
        });
        mViewModel.getImages(movie.getId());
        return mBinding.getRoot();
    }


    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);


    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        mBinding = null;
    }
}

File: app/src/main/java/com/example/cinemates/ui/details/MediaImagesFragment.kt
package com.example.cinemates.ui.details


import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.navigation.fragment.navArgs
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.StaggeredGridLayoutManager
import com.example.cinemates.data.remote.response.image.ImageType
import com.example.cinemates.databinding.FragmentListingItemsBinding
import com.example.cinemates.ui.adapter.ImageAdapter
import com.google.android.material.bottomsheet.BottomSheetDialogFragment

/**
 *@author Antonio Di Nuzzo (Indisparte)
 */
class MediaImagesFragment : BottomSheetDialogFragment() {
    private var _binding: FragmentListingItemsBinding? = null
    private val binding: FragmentListingItemsBinding
        get() = _binding!!
    private lateinit var adapter: ImageAdapter
    private val args: MediaImagesFragmentArgs by navArgs()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        adapter = ImageAdapter()
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        // Inflate the layout for this fragment
        _binding = FragmentListingItemsBinding.inflate(inflater, container, false)


        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        setupRecyclerView()

    }

    private fun setupRecyclerView() {
        binding.layoutRecyclerView.apply {
            val images = args.images.asList()
            if(images.isNotEmpty()) {
                val imageType = images[0].imageType
                if (imageType == ImageType.BACKDROP) {
                    recyclerView.layoutManager =
                        LinearLayoutManager(requireContext())
                } else if (imageType == ImageType.POSTER) {
                    recyclerView.layoutManager =
                        GridLayoutManager(requireContext(), 3)
                }
                recyclerView.adapter = adapter
                adapter.items = images
            }
            recyclerView.adapter = adapter
            adapter.items = images
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }

}